src/tendencies/vertical_diffusion_boundary_layer.jl

```diff
@@ -323,25 +310,25 @@ function get_surface_fluxes!(Y, p, t, colidx, ::VerticalDiffusion)
     end
 
     if diffuse_momentum(p.atmos.vert_diff)
-        ρτxz = sfc_conditions[colidx].ρτxz
-        ρτyz = sfc_conditions[colidx].ρτyz
-        ρ_1 = Fields.level(Y.c.ρ[colidx], 1)
-        @. dif_flux_uₕ[colidx] =
-            Geometry.Contravariant3Vector(surface_normal[colidx]) ⊗
-            Geometry.Covariant12Vector(
-                Geometry.UVVector(ρτxz / ρ_1, ρτyz / ρ_1),
-            )
-        @. dif_flux_uₕ_bc[colidx] = -dif_flux_uₕ[colidx]
+        if isnothing(p.surface_scheme)
+            @. dif_flux_ρuₕ[colidx] *= 0
+        else
+            @. dif_flux_ρuₕ[colidx] =
+                Geometry.WVector(FT(1)) ⊗ Geometry.UVVector(
```

Comment:
Why could we remove the `Contravarian3Vector` and `Covariant12Vector`?

Reply:
For the same reason we don't need any conversion [here](https://github.com/CliMA/ClimaAtmos.jl/blob/01ae1f01f9435636b0fb42aa6d8841683585fa86/src/tendencies/vertical_diffusion_boundary_layer.jl#L410); the finite difference operators automatically convert axis tensors to the required types.